### ISSUE TITLE
[Python docs feedback] issues in code snippet from 'Develop your app'

### ISSUE THREAD

**Issue by guillaumeklimek** [docker/docker.github.io#12518]
### Problem description

There is 2 issues in the code snippet from 'Develop your app' in Python:

- 1/ There is no db password creation so it must be empty in the 3 connectors instantiations
- 2/ There is no way to save widgets (as mentioned bellow the snippet)

### Problem location

I saw a problem on the following URL: https://docs.docker.com/language/python/develop/#connect-the-application-to-the-database

### Project version(s) affected

It's about the code snippet.

### Suggestions for a fix

**1/ Fix the password issue**

```python
mydb = mysql.connector.connect(
   host="mysqldb",
   user="root",
   password="",  # TODO must be empty in each call
   database="inventory"
)
```

**2/ Add a way to save widgets using get_widgets()**

To keep it simple, I decided to:
- add the `request` object from `flask` (`from flask import request`)
- add the POST methods to the route decorator (`methods=['POST']`)
- read the data contained in the POST request with the request object (`request.form['name']`)
- insert the data in the database (before the fetch part)

**Note:** it breaks many best practices. In a REST API, it would be better to have 2 different routes, one to POST (create data) and the other one to GET (read data).

```python
from flask import request

@app.route('/widgets',  methods=['POST'])
def get_widgets():
    mydb = mysql.connector.connect(
        host="mysqldb",
        user="root",
        password="",
        database="inventory"
    )
    cursor = mydb.cursor()

    # POST part
    name = request.form['name']
    description = request.form['description']
    cursor.execute(f"INSERT INTO widgets VALUES ('{name}', '{description}')")

    # GET part
    cursor.execute("SELECT * FROM widgets")

    row_headers = [x[0] for x in cursor.description]  # this will extract row headers

    results = cursor.fetchall()
    json_data = []
    for result in results:
        json_data.append(dict(zip(row_headers, result)))

    cursor.close()
```

Best Regards,
Guillaume Klimek

**Comment by guillaumeklimek** [comment]
But,

The rest of tutorial will match with the current snippet (even if it's not very interactive).

### Suggestion

Change the way of testing the mysqldb connector.

### Another problem

Wrong port in the last curl command: `curl --request GET --url http://localhost:8080/widgets`. Port is `5000`.

**Comment by StefanScherer** [comment]
Thanks @guillaumeklimek, we're working on a fix in #12559.
